src/mechanics/constraintsPde/ConstraintCompanion.cpp

```diff
@@ -48,29 +50,25 @@ std::vector<Equation> Component(const Groups::Group<NodeSimple>& nodes, std::vec
 Equation Direction(const NodeSimple& node, Eigen::VectorXd direction, RhsFunction rhs)
 {
 
-    assert(direction != Eigen::VectorXd::Zero(direction.rows()) && "the direction vector contains only zeros");
+    int maxComponentIndex = -1;
+    if (direction.cwiseAbs().maxCoeff(&maxComponentIndex) < 1.e-6)
+        throw Exception(__PRETTY_FUNCTION__,
+                        "Your direction vector is composed of zeros only! The direction is unspecified!");
+
+    assert(maxComponentIndex > -1 && maxComponentIndex < direction.rows());
```

Comment:
You don't trust `Eigen`, right? :P

Reply:
I don't trust anybody!!! Especially not myself. Not sure if this was just a relict from previous commits. Anyhow, I removed it.